### PR TITLE
Fix a memory leak in EC_GROUP_get_ecparameters()

### PR DESCRIPTION
**Pull request by rhenium**
The variable 'buffer', allocated by EC_POINT_point2buf(), isn't
free'd on the success path.

### CODE REVIEW

**Comment by richsalz** [comment]
+1

**Comment by ekasper** [comment]
A memory leak on a _success_ path really should be caught by tests... Would you mind adding a test to ectest?

**Comment by mattcaswell** [comment]
Ping @rhenium - are you able to respond to @ekasper's comment? I'd like to get this in for the 1.1.0 release.

**Comment by rhenium** [comment]
I am sorry, I've missed your comment. Added to prime_field_tests(), but I'm not sure if I'm on the right course. Feel free to discard if not.

File: util/libcrypto.num [hunk 1]
```diff
@@ -4193,3 +4193,7 @@ EVP_PKEY_get1_tls_encodedpoint          4139	1_1_0	EXIST::FUNCTION:
 CT_POLICY_EVAL_CTX_set1_cert            4140	1_1_0	EXIST::FUNCTION:CT
 CT_POLICY_EVAL_CTX_set1_issuer          4141	1_1_0	EXIST::FUNCTION:CT
 CT_POLICY_EVAL_CTX_set_shared_CTLOG_STORE 4142	1_1_0	EXIST::FUNCTION:CT
+ECPKPARAMETERS_free                     4143	1_1_0	EXIST::FUNCTION:EC
+ECPARAMETERS_free                       4144	1_1_0	EXIST::FUNCTION:EC
+ECPKPARAMETERS_new                      4145	1_1_0	EXIST::FUNCTION:EC
+ECPARAMETERS_new                        4146	1_1_0	EXIST::FUNCTION:EC
```
Comment (mattcaswell) [inline util/libcrypto.num]: These new functions need to be documented in the pod files.

File: test/ectest.c [hunk 2]
```diff
@@ -876,6 +876,23 @@ static void prime_field_tests(void)
...
+        if (ecparameters == NULL)
+            ABORT;
+        group2 = EC_GROUP_new_from_ecparameters(ecparameters);
+        if (group2 == NULL)
+            ABORT;
+        if (EC_GROUP_cmp(group, group2, ctx))
+            ABORT;
+
+        EC_GROUP_free(group2);
+        ECPARAMETERS_free(ecparameters);
```
Comment (mattcaswell) [inline test/ectest.c]: This really has nothing to do with prime field testing. I would add this test as a separate function (e.g. `paramater_tests()` or something), and add a call to it after the `internal_curve_test()` call in `main`.

File: util/libcrypto.num (hunk 1 above)
Comment (richsalz) [inline util/libcrypto.num]: they are auto-generated ASN1 functions, which we haven't been documenting.

Comment (mattcaswell) [inline util/libcrypto.num]: Really why not? Perhaps we could get away with a generic page for all the ASN1 functions (if so I agree that would be out of scope for this PR)

Comment (richsalz) [inline util/libcrypto.num]: Because we have not made the ASN1 stuff officially part of the public API.

Comment (mattcaswell) [inline util/libcrypto.num]: Huh? Yeah...I think we have!

**Comment by rhenium** [comment]
Updated. I added these auto-generated alloc functions in doc/crypto/X509_dup.pod as other functions declared by DECLARE_ASN1_ALLOC_FUNCTIONS are already in it.

**Comment by mattcaswell** [comment]
+1

**Comment by mattcaswell** [comment]
@richsalz - please reconfirm your plus one.

**Comment by richsalz** [comment]
+1.  you merging, @mattcaswell ?

**Comment by mattcaswell** [comment]
Merged. Thanks.

**Pull request closed without merging**